raiwidgets/tests/test_cohort.py

```diff
@@ -141,7 +150,343 @@ def test_cohort_filter_serialization_include_exclude_methods(self, method):
+                      "test data"):
+            cohort_filter = \
+                CohortFilter(method=CohortFilterMethods.METHOD_INCLUDES,
+                             arg=['mid'],
+                             column='type')
+            cohort_filter._validate_with_test_data(
+                test_data=test_data,
+                target_column="target",
+                categorical_features=['type'],
+                is_classification=True)
+
+
 class TestCohort:
+    def test_cohort_configuration_validations(self):
+        with pytest.raises(
+            UserConfigValidationException,
+            match="Got unexpected type <class 'int'> for cohort name. "
+                  "Expected string type."):
+            Cohort(name=1)
+
+        with pytest.raises(
+            UserConfigValidationException,
+            match="Got unexpected type <class 'list'> for cohort filter. "
+                  "Expected CohortFilter type"):
+            cohort = Cohort(name="Cohort New")
+            cohort.add_cohort_filter(cohort_filter=[])
+
+    def test_cohort_validate_with_test_data(self):
+        cohort_filter_1 = \
+            CohortFilter(method=CohortFilterMethods.METHOD_LESS,
+                         arg=[65], column='age')
+        cohort_1 = Cohort(name="Cohort New")
+        cohort_1.add_cohort_filter(cohort_filter_1)
+        test_data = pd.DataFrame(data=[[23, 'X'], [25, 'Y']],
+                                 columns=["age", "target"])
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="The test_data should be a pandas DataFrame"):
+            cohort_1._validate_with_test_data(
+                test_data=[], target_column='income',
+                categorical_features=[])
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="The target_column should be string."):
+            cohort_1._validate_with_test_data(
+                test_data=test_data,
+                target_column=1,
+                categorical_features=[])
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="The target_column fake_target "
+                      "was not found in test_data."):
+            cohort_1._validate_with_test_data(
+                test_data=test_data,
+                target_column="fake_target",
+                categorical_features=[])
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="Expected a list type for "
+                      "categorical columns."):
+            cohort_1._validate_with_test_data(
+                test_data=test_data,
+                target_column="target",
+                categorical_features={})
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="All entries in categorical_features "
+                      "need of string type."):
```

Comment:
need of string type -> need to be of string type (?)

Reply:
Done